src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -49,6 +57,11 @@ public function normalize($object, $format = null, array $context = array())
         }
 
         $format = isset($context[self::FORMAT_KEY]) ? $context[self::FORMAT_KEY] : $this->format;
+        $timezone = $this->getTimezone($context);
+
+        if (null !== $timezone) {
+            $object = (new \DateTime('@'.$object->getTimestamp()))->setTimezone($timezone);
```

Comment:
The thing is to not use the second argument here actually, because as specified on the manual entry:

> The $timezone parameter and the current timezone are ignored when the $time parameter either is a UNIX timestamp (e.g. @946684800) or specifies a timezone (e.g. 2010-01-28T15:00:00+02:00).

— http://php.net/manual/en/datetime.construct.php#refsect1-datetime.construct-parameters

 on the contrary of `\DateTime::setTimeZone()` which will properly set the timezone and update the date accordingly.

See the following:

```php
>>> new \DateTime('@1480546800')
=> DateTime {#166
     +"date": "2016-11-30 23:00:00.000000",
     +"timezone_type": 1,
     +"timezone": "+00:00",
   }
>>> new \DateTime('@1480546800', new \DateTimeZone('Japan'))
=> DateTime {#173
     +"date": "2016-11-30 23:00:00.000000",
     +"timezone_type": 1,
     +"timezone": "+00:00",
   }
>>> (new \DateTime('@1480546800'))->setTimeZone(new \DateTimeZone('Japan'))
=> DateTime {#174
     +"date": "2016-12-01 08:00:00.000000",
     +"timezone_type": 3,
     +"timezone": "Japan",
   }
>>>
```